bayesian/api_v1.py

```diff
@@ -919,35 +924,53 @@ def post():
 
             manifests.append(manifest)
 
-        try:
-            req = StackAnalysisRequest(
-                id=request_id,
-                submitTime=str(dt),
-                requestJson={'manifest': manifests},
-            )
-            rdb.session.add(req)
-            rdb.session.commit()
-        except SQLAlchemyError as e:
-            raise HTTPError(500, "Error inserting log for request {t}".format(t=request_id)) from e
-
         data = {'api_name': 'stack_analyses',
                 'user_email': decoded.get('email', 'bayesian@redhat.com'),
                 'user_profile': decoded}
-        args = {'external_request_id': request_id, 'ecosystem': ecosystem, 'data': data}
+        args = {'external_request_id': request_id,
+                'ecosystem': ecosystem, 'data': data}
 
         try:
             api_url = current_app.config['F8_API_BACKBONE_HOST']
 
             d = DependencyFinder()
-            deps = d.execute(args, rdb.session)
+            deps = d.execute(args, rdb.session, manifests)
             deps['external_request_id'] = request_id
 
             _session.post('{}/api/v1/stack_aggregator'.format(api_url), json=deps)
             _session.post('{}/api/v1/recommender'.format(api_url), json=deps)
         except Exception as exc:
             raise HTTPError(500, ("Could not process {t}.".format(t=request_id))) from exc
-
-        return {"status": "success", "submitted_at": str(dt), "id": str(request_id)}
+        if persist:
+            if sid:
+                try:
+                    result = rdb.session.query(StackAnalysisRequest)\
```

Comment:
You can also look at http://docs.sqlalchemy.org/en/latest/dialects/postgresql.html?highlight=conflict#insert-on-conflict-upsert

Reply:
Thanks @miteshvp, I will do the changes.